spec/controllers/track_controller_spec.rb

```diff
@@ -16,295 +16,367 @@
-  before(:each) do
-    @ir = mock_model(InfoRequest, :url_title => 'myrequest',
-                     :title => 'My request')
-    @track_thing = mock_model(TrackThing, :save! => true,
-                              :params => {},
-                              :track_medium= => nil,
-                              :tracking_user_id= => nil)
-    allow(TrackThing).to receive(:create_track_for_request).and_return(@track_thing)
-    allow(TrackThing).to receive(:create_track_for_search_query).and_return(@track_thing)
-    allow(TrackThing).to receive(:find_existing).and_return(nil)
-    allow(InfoRequest).to receive(:find_by_url_title!) do |url_title|
-      if url_title == "myrequest"
-        @ir
-      else
-        raise ActiveRecord::RecordNotFound.new("Not found")
-      end
+    it "should set no-cache headers on the login redirect" do
+      get :track_request, :url_title => ir.url_title, :feed => 'track'
+      expect(response.headers["Cache-Control"]).
+        to eq('no-cache, no-store, max-age=0, must-revalidate')
+      expect(response.headers['Pragma']).to eq('no-cache')
+      expect(response.headers['Expires']).to eq('0')
     end
 
-    @user = mock_model(User)
-    allow(User).to receive(:find).and_return(@user)
-    allow(@user).to receive(:locale).and_return("en")
-    allow(@user).to receive(:receive_email_alerts).and_return(true)
-    allow(@user).to receive(:url_name).and_return("bob")
-  end
+    it "should save a request track and redirect if you are logged in" do
+      session[:user_id] = user.id
+      allow(TrackThing).to receive(:create_track_for_request).and_return(track_thing)
+      expect(track_thing).to receive(:save).and_call_original
+      get :track_request, :url_title => ir.url_title, :feed => 'track'
+      expect(response).to redirect_to(:controller => 'request',
+                                      :action => 'show', :url_title => ir.url_title)
+    end
 
-  it "should require login when making new track" do
-    get :track_request, :url_title => @ir.url_title, :feed => 'track'
-    expect(response).to redirect_to(:controller => 'user',
-                                    :action => 'signin',
-                                    :token => get_last_post_redirect.token)
+    it "should 404 for non-existent requests" do
+      session[:user_id] = user.id
+      expect {
+        get :track_request, :url_title => "hjksfdhjk_louytu_qqxxx", :feed => 'track'
+      }.to raise_error(ActiveRecord::RecordNotFound)
+    end
   end
 
-  it "should set no-cache headers on the login redirect" do
-    get :track_request, :url_title => @ir.url_title, :feed => 'track'
-    response.headers["Cache-Control"].should == 'no-cache, no-store, max-age=0, must-revalidate'
-    response.headers['Pragma'].should == 'no-cache'
-    response.headers['Expires'].should == '0'
-  end
+  describe "when making a search track" do
+    let(:track_thing) { FactoryGirl.create(:search_track,
```

Comment:
I don't disagree with the rule as it happens – you can still do:

```ruby
let(:blah) do
  # ...
end
```

If you rebase on develop you should get the updated config that disables hound checks by default, other than the ones we've explicitly enabled though :)